packages/react-aria-components/src/Popover.tsx

```diff
@@ -170,6 +173,7 @@ function PopoverInner({state, isExiting, UNSTABLE_portalContainer, ...props}: Po
         ref={ref}
         slot={props.slot || undefined}
         style={style}
+        dir={props.dir}
```

Comment:
popovers are portaled outside which means they don't get the direction of the locale causing the keyboard navigation to behave incorrectly for datepickers. in v3, we wrap the popovers in a provider so this doesn't happy to change the implementation if there's a better way to do this